### PR TITLE
Refactor Directory structure

### PR DESCRIPTION
**Pull request by YJDoc2**
Apologies for the big PR, but major of the changes are moved files.
This PR does some needed refactoring of directory structure, which will make it easier to understand and work on youki.

Major Changelog :
- created a dir integration-tests which will hold all integration tests, including oci-tests and its rust port, and later containerd etc.
- moved rust-integration-tests, runtimetest and test_framework from /crates to integration_tests/rust-integration-tests, as those do not belong to youki crates, and are better fit in the integration tests
- added scripts dir, where all the scripts including build script and script to run the integration tests will reside, this will make it easier to modify and run tests and build from a single point
- moved the oci-tests submodule from /integration_test to /integration_tests/oci-runtime-tests

This PR will form the first step for #601 , and does not complete it. I made this PR, as even by itself it is quite big, and I did not want to make the review more hard. After this is merged, the next steps in separate PRs will be :
- Improve scripts and makefiles . This PR adds scripts and makefiles just to make it work with CI. They are still a bit cumbersome to use manually. I will make another PR which should make it easier to manually use.
- Update the docs. Even though no code is changed, the docs need to be updated for new directory structure and scripts. This will also need its own separate PR.

When testing on my own fork, I noticed that the oci-integration-tests were failing, but not because of changes I made, but because one of the go tests was failing. Adding its link here, in case the same happens in this PR : https://github.com/YJDoc2/youki/runs/5091082926?check_suite_focus=true

<a href="https://gitpod.io/#https://github.com/containers/youki/pull/694"></a>

### CODE REVIEW

**Comment by YJDoc2** [comment]
It appears that the same issue is also failing this CI , with same 3 capabilites :
```
...
not ok 53 - unexpected bounding capability CAP_PERFMON not set
not ok 54 - unexpected bounding capability CAP_BPF not set
not ok 55 - unexpected bounding capability CAP_CHECKPOINT_RESTORE not set
ok 56 - expected effective capability CAP_CHOWN set
...
ok 93 - unexpected effective capability CAP_AUDIT_READ not set
not ok 94 - unexpected effective capability CAP_PERFMON not set
not ok 95 - unexpected effective capability CAP_BPF not set
not ok 96 - unexpected effective capability CAP_CHECKPOINT_RESTORE not set
ok 97 - expected inheritable capability CAP_CHOWN set
...
ok 175 - unexpected permitted capability CAP_AUDIT_READ not set
not ok 176 - unexpected permitted capability CAP_PERFMON not set
not ok 177 - unexpected permitted capability CAP_BPF not set
not ok 178 - unexpected permitted capability CAP_CHECKPOINT_RESTORE not set
ok 179 - expected ambient capability CAP_CHOWN set
...

```

**Comment by YJDoc2** [comment]
Hey, no issues, take your time :)
I am working on why the integration tests are failing, after that I'll also solve the conflicts, so if there are any other commits by then and other conflicts, those will get solved as well.
Thanks :)

File: integration_tests/rust-integration-tests/integration_test/Cargo.toml [hunk 1]
```diff
@@ -7,12 +7,12 @@ edition = "2021"
...
 flate2 = "1.0"
-libcgroups = { path = "../libcgroups" }
-libcontainer = { path = "../libcontainer" }
+libcgroups = { path = "../../../crates/libcgroups" }
+libcontainer = { path = "../../../crates/libcontainer" }
 log = { version = "0.4", features = ["std"] }
 nix = "0.23.1"
 num_cpus = "1.13"
-oci-spec = { git = "https://github.com/containers/oci-spec-rs", rev = "54c5e386f01ab37c9305cc4a83404eb157e42440" }
+oci-spec = "0.5.3"
```
Comment (utam0k) [inline integration_tests/rust-integration-tests/integration_test/Cargo.toml:15]: 👍

File: scripts/build.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#! /bin/sh
+
+ROOT=$(git rev-parse --show-toplevel)
+
+cd ${ROOT}/crates
+make release
```
Comment (utam0k) [inline scripts/build.sh:6]: I think it would be better the default is debug. WDYT?

File: scripts/rust_integration_tests.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#! /bin/sh
+
```
Comment (utam0k) [inline scripts/rust_integration_tests.sh:2]: we need to put `set -eu` at the beginning of all scripts.

**Comment by utam0k** [comment]
It would be nice if the make command could work from the top of the directory, is that possible?

File: scripts/build.sh (hunk 2 above)
Comment (YJDoc2) [reply scripts/build.sh:6]: I think in general that would be a good idea, but for now, this is only used by the integration_tests, which need the youki binary to be in release mode. As mentioned in the point 1 in the PR comment, I plan to improve scripts and makefiles in a separate PR,  where I will make the default be the debug mode.

File: scripts/rust_integration_tests.sh (hunk 3 above)
Comment (YJDoc2) [reply scripts/rust_integration_tests.sh:2]: :+1:

**Comment by YJDoc2** [comment]
Hey @utam0k finally the CI is done :sweat_smile: 
Please take a look :)

**Comment by utam0k** [comment]
@YJDoc2 Can I ask you to update the README and resolve conflicts since the path to `build.sh` has changed?

**Comment by YJDoc2** [comment]
Hey @utam0k I have updated the readme and docs tutorial, please check
Also Currently there are conflicts, but what I feel is :
If we merge this one first, then there will be a lot of issues for PRs which are open like #758 etc.
I do not mind waiting till they are merged, and then resolving conflicts on my side.
Let me know :+1:

**Comment by YJDoc2** [comment]
Hey, the conflicts are resolved, the CI is passing, and currently there are no other PRs open!
Please take a look :+1: :smile:

**Review by utam0k — APPROVED** [review]
💯

**Pull request merged**